src/cli/cli.cpp

```diff
@@ -2062,6 +2065,66 @@ void Interpreter::ProcessNetif(uint8_t aArgsLength, char *aArgs[])
 }
 #endif
 
+#if OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE
```

Comment:
What's the motivation for using `OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE` feature flag?

Reply:
`otUdpGetSockets` is defined only if `OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE=1`.

Reply:
Yes, but why? 😄 @bukepo , thoughts?

Reply:
I noticed the same in `udp_api.cpp` (though it does not explicitly mention this in the `udp.h` documentation of this API). I guess when it was added it was only used/intended when `PLATFORM_UDP` feature was enabled. 

I also favor that we do not restrict this API to this config and make available in general.

Reply:
Abtin helped answer this question, and yes, agree not to restrict this API.

Reply:
Removed this flag requirement from API/CLI.